frontend_lib/src/customHooks.jsx

```diff
@@ -0,0 +1,40 @@
+import React, { useEffect } from 'react'
+import { CUSTOM_EVENT } from './customEvent'
+
+/* FIXME - ML - 2022-10-20 - Export useLifecycle hook when it's usage will be appropriate.
+  * This hook is designed to bring a better syntax to the useEffect hook when it is
+  * used to replace class component's didMount and willUnmount (passing [] as deps)
+  * https://github.com/tracim/tracim/issues/5954
+*/
+const useLifecycle = ({ mountedEvent, unmountedEvent }) => {
+  useEffect(() => {
+    mountedEvent()
+    return unmountedEvent
+  }, [])
+}
+
+export const usePublishLifecycle = (componentName, data, dispatchEvent) => {
+  useLifecycle({
+    mountedEvent: () => dispatchEvent(CUSTOM_EVENT.TRACIM_COMP_MOUNTED(componentName), data),
+    unmountedEvent: () => dispatchEvent(CUSTOM_EVENT.TRACIM_COMP_UNMOUNTED(componentName), data)
+  })
+}
+
+export const withUsePublishLifecycle = (WrappedComponent, componentName) => {
```

Comment:
I'd like a little comment here to explain that this is a HOC to allow using custom hook inside class component

Reply:
done